### PR TITLE
ENH: Re-enable VXE from build targets for sin/cos 

### PR DESCRIPTION
**Pull request by r-devulap**
As discussed in the optimization meeting, @seiko2plus wanted to split VSX and VXE into two separate PR's. 

For clarification, SIMD optimizations for sine and cosine functions on both ppc64 and z/Architecture (IBM Z) were disabled by #25781 to bypass CI tests. This PR aims to re-enable optimizations for z/Architecture after addressing the following runtime errors, while #27627 re-enabled ppc64 optimizations.
```Bash

numpy/_core/tests/test_umath_accuracy.py:84: AssertionError
=========================== short test summary info ============================
FAILED numpy/_core/tests/test_umath.py::TestSpecialFloats::test_sincos_values[f] - AssertionError: 
FAILED numpy/_core/tests/test_umath.py::TestSpecialFloats::test_sincos_errors[inf-f-sin] - AssertionError: FloatingPointError not raised by sin
FAILED numpy/_core/tests/test_umath.py::TestSpecialFloats::test_sincos_errors[inf-f-cos] - AssertionError: FloatingPointError not raised by cos
FAILED numpy/_core/tests/test_umath.py::TestSpecialFloats::test_sincos_errors[-inf-f-sin] - AssertionError: FloatingPointError not raised by sin
FAILED numpy/_core/tests/test_umath.py::TestSpecialFloats::test_sincos_errors[-inf-f-cos] - AssertionError: FloatingPointError not raised by cos
FAILED numpy/_core/tests/test_umath.py::TestAVXFloat32Transcendental::test_sincos_float32 - AssertionError: Arrays are not almost equal up to 2 ULP (max difference is ...
FAILED numpy/_core/tests/test_umath.py::TestAVXFloat32Transcendental::test_strided_float32 - AssertionError: Arrays are not equal to 2 ULP (max is 1.05785e+09)
FAILED numpy/_core/tests/test_umath_accuracy.py::TestAccuracy::test_validate_fp16_transcendentals[cos] - AssertionError: Arrays are not almost equal up to 1 ULP (max difference is ...
FAILED numpy/_core/tests/test_umath_accuracy.py::TestAccuracy::test_validate_fp16_transcendentals[sin] - AssertionError: Arrays are not almost equal up to 1 ULP (max difference is ...
= 9 failed, 22213 passed, 1559 skipped, 26186 deselected, 7 xfailed, 2 xpassed in 555.12s (0:09:15) =
````

### CODE REVIEW

File: numpy/_core/src/umath/loops_trigonometric.dispatch.cpp [hunk 1]
```diff
@@ -151,23 +154,36 @@ simd_sincos_f32(const float *src, npy_intp ssrc, float *dst, npy_intp sdst,
...
             }
         }
         if (!hn::AllTrue(f32, simd_mask)) {
+            static_assert(hn::MaxLanes(f32) <= 16,
+                          "The following fallback is not applicable for "
+                          "SIMD widths larger than 512 bits, or for scalable "
+                          "SIMD in general.");
             npy_uint64 simd_maski;
-            hn::StoreMaskBits(f32, simd_mask, (uint8_t*)&simd_maski);
+            hn::StoreMaskBits(f32, simd_mask, (uint8_t *)&simd_maski);
```
Comment (seiko2plus) [inline numpy/_core/src/umath/loops_trigonometric.dispatch.cpp:180]: A better approach is needed for libc fallback to properly handle scalable extensions. One potential improvement could be introducing a new intrinsic, such as `bool TestBit(V, int pos)`, which would also be endian-friendly. For now, I applied a quick fix below to address a big-endian bug, as passing a `uint64_t` as a `uint8_t` array obviously lead to accessing garbage bytes instead.

File: numpy/_core/meson.build [hunk 2]
```diff
@@ -953,6 +953,7 @@ foreach gen_mtargets : [
     [
       AVX512_SKX, [AVX2, FMA3],
       NEON_VFPV4,
+      VXE2, VXE
```
Comment (seiko2plus) [inline numpy/_core/meson.build:956]: need a re-base maybe? ppc64 features `VSX4, VSX3, VSX2` are missing???

**Comment by seiko2plus** [comment]
@r-devulap, Oh, my editor automatically reformatted the entire source code based on the NumPy clang-format configuration. Would you prefer that I revert these changes, or is this formatting acceptable for you?

**Comment by seiko2plus** [comment]
All CI tests for s390x have passed.

**Comment by seberg** [comment]
@r-devulap if you are happy about the changes please just put it in (I don't think the reformatting matters too much -- it's not that much).
2.2.0 is going to be branched very soon, it it seems reasonable to want to have this in.

File: numpy/_core/meson.build (hunk 2 above)
Comment (r-devulap) [reply numpy/_core/meson.build]: Yeah looks like I made a mess of the rebase :/. It should be fixed now.

**Comment by r-devulap** [comment]
> @r-devulap, Oh, my editor automatically reformatted the entire source code based on the NumPy clang-format configuration. Would you prefer that I revert these changes, or is this formatting acceptable for you?

Yeah no worries. This looks better anyways.

**Comment by r-devulap** [comment]
> @r-devulap if you are happy about the changes please just put it in (I don't think the reformatting matters too much -- it's not that much). 2.2.0 is going to be branched very soon, it it seems reasonable to want to have this in.

yeah makes sense. I had to fix some rebase problems. But once the CI passes it should be good to go in.

**Comment by seberg** [comment]
Let's put this in then, thanks @r-devulap and @seiko2plus.

**Comment by Sunny-Anand** [comment]
Not sure if this is having any impact here. https://github.com/google/highway/issues/2409

**Comment by Sunny-Anand** [comment]
I can confirm that trying to install numpy>= 2.0.0 for python3.10 breaks on UBI-8 image.

**Pull request merged**